src/components/trade/TradesTable/index.tsx

```diff
@@ -0,0 +1,173 @@
+      <th>
+        Limit price
+        <Icon icon={faExchangeAlt} onClick={tableContext.invertPrice} />
+      </th>
+      <th>
+        Execution Price
+        <Icon icon={faExchangeAlt} onClick={tableContext.invertPrice} />
+      </th>
+      <th>Tx Hash</th>
+      <th>Surplus</th>
+      <th>Trade Time</th>
+    </tr>
+  )
+}
+
+export type Props = {
+  header?: JSX.Element
+  className?: string
+  numColumns?: number
+  data: Array<Order>
+}
+
+const TitleUid = styled(RowWithCopyButton)`
+  color: ${({ theme }): string => theme.grey};
+  font-size: ${({ theme }): string => theme.fontSizeDefault};
+  font-weight: ${({ theme }): string => theme.fontNormal};
+  display: flex;
+  align-items: center;
+`
+
+export const TradesTable = ({ header, className, numColumns, data }: Props): JSX.Element => {
+  const tableContext = React.useContext(TradesTableContext)
+  const [rows, setRows] = React.useState([{}])
+
+  React.useEffect((): void => {
+    // for each entry in the data array invert the values of
+    // the limit price and execution price if toggled
+    const _rows: Array<
+      Order & {
+        limitPrice: string
+        executionPrice: string
+      }
+    > = data.map((order): any => {
+      return {
+        ...order,
+        limitPrice:
+          order.buyToken && order.sellToken
+            ? constructPrice({
+                isPriceInverted: tableContext.isPriceInverted,
+                order,
+                data: {
+                  numerator: {
+                    amount: order.sellAmount,
+                    token: order.sellToken,
+                  },
+                  denominator: {
+                    amount: order.buyAmount,
+                    token: order.buyToken,
+                  },
+                },
+              })
+            : '-',
+        executionPrice:
+          order.buyToken && order.sellToken
+            ? constructPrice({
+                isPriceInverted: tableContext.isPriceInverted,
+                order,
+                data: {
+                  numerator: {
+                    amount: order.buyAmount,
+                    token: order.buyToken,
+                  },
+                  denominator: {
+                    amount: order.sellAmount,
+                    token: order.sellToken,
+                  },
+                },
+              })
+            : '-',
+      }
+    })
+    setRows(_rows)
+  }, [data, tableContext.isPriceInverted])
+
+  return (
+    <SimpleTable
+      body={
+        <>
+          {rows.length > 0 &&
+            rows.map(
+              (order: Order & { limitPrice: string; executionPrice: string }, i) =>
+                order.uid && (
+                  <tr key={i}>
+                    <td>
+                      <TitleUid textToCopy={order.uid} contentsToDisplay={order.shortId} />
+                    </td>
+                    <td className={i % 2 === 1 ? 'long' : 'short'}>{i % 2 === 1 ? 'Buy' : 'Sell'}</td>
```

Comment:
It's slightly different in the design.
I'm fine to address in another PR though. You can for now leave out the className thing.
And if you want to test inverting buy/sell, just change the data in the storybook :)